### PR TITLE
Add SilverStripe 4 support

### PR DESCRIPTION
**Pull request by ishannz**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -1,22 +1,35 @@
...
-	"require": {
-		"silverstripe/framework": "~3.1"
-	},
-	"extra": {
-		"installer-name": "leafletfield"
-	}
+    "name": "nzta/googlemap-leafletfield",
+    "description": "Provides a form field type allowing users to draw features on a map, the features are stored as geoJSON objects in a single database field",
+    "type": "silverstripe-vendormodule",
+    "homepage": "https://github.com/nzta/googlemap-leafletfield",
```
Comment (pjayme) [inline composer.json:5]: Would be good to add a license here of `BSD-3` as well as the actual license file itself

File: src/LeafletField.php [hunk 2]
```diff
@@ -1,19 +1,32 @@
 <?php
```
Comment (pjayme) [inline src/LeafletField.php]: Might be good to insert this into a `Forms` folder ie. `src/forms/LeafletField.php` and update the namespaces too :)

File: src/LeafletField.php [hunk 3]
```diff
@@ -86,146 +106,156 @@ public function Field($properties = array())
...
+
+        Requirements::css('//cdnjs.cloudflare.com/ajax/libs/leaflet/1.0.3/leaflet.css');
+        Requirements::css('//cdnjs.cloudflare.com/ajax/libs/leaflet.draw/0.4.9/leaflet.draw.css');
+        Requirements::css('nzta/googlemap-leafletfield: client/css/LeafletField.css');
     }
 
     /**
-     * Return the L.Control.Draw options.
-     * @return Array
+     * {@inheritdoc}
```
Comment (pjayme) [inline src/LeafletField.php]: Dont think this should say `@inheritdoc`?

File: src/LeafletField.php [hunk 4]
```diff
@@ -86,146 +106,156 @@ public function Field($properties = array())
...
+            $this->geometryField->setValue($value);
+        }
+
+        return $this;
     }
 
     /**
-     * Return the draw options as a json string.
-     * @return String
+     * {@inheritdoc}
```
Comment (pjayme) [inline src/LeafletField.php]: Dont think this should say `@inheritdoc`?

File: src/LeafletField.php [hunk 5]
```diff
@@ -55,6 +69,7 @@ public function __construct($name, $title = null, DataObject $data)
 
     /**
      * Set up child hidden fields.
+     *
      * @return FieldList
      */
     public function setupChildren($name)
```
Comment (pjayme) [inline src/LeafletField.php]: needs a `@param` for `$name`

File: composer.json (hunk 1 above)
Comment (ishannz) [reply composer.json:5]: done

File: src/LeafletField.php (hunk 2 above)
Comment (ishannz) [reply src/LeafletField.php]: done

File: src/LeafletField.php (hunk 5 above)
Comment (ishannz) [reply src/LeafletField.php]: done

File: src/LeafletField.php (hunk 3 above)
Comment (ishannz) [reply src/LeafletField.php]: No idea what is it and how it comes. change back to array

**Pull request merged**